api/src/certification/shared/domain/models/CertificationCourse.js

```diff
@@ -305,11 +305,15 @@ class CertificationCourse {
   }
 
   getNumberOfChallenges() {
-    if (this.isV3()) {
-      return this._numberOfChallenges;
+    if (_.isUndefined(this._numberOfChallenges)) {
```

Comment:
De ce que je vois cette fonction `getNumberOfChallenges` est appelee uniquement par le serializer des deux usecases ou tu as code le `setNumberOfChallenges` ; donc il semble quel 'on peut simplifier ce getter que je trouve en l'etat un peu complexe 😅 

Ceci devrait suffir du coup non ?

```suggestion
    return this._numberOfChallenges || 0;
```